airflow/executors/kubernetes_executor.py

```diff
@@ -689,9 +689,14 @@ def _change_state(self, key: TaskInstanceKey, state: Optional[str], pod_id: str,
         self.event_buffer[key] = state, None
 
     def try_adopt_task_instances(self, tis: List[TaskInstance]) -> List[TaskInstance]:
-        tis_to_flush = [ti for ti in tis if not ti.queued_by_job_id]
-        scheduler_job_ids = {ti.queued_by_job_id for ti in tis}
-        pod_ids = {ti.key: ti for ti in tis if ti.queued_by_job_id}
+        scheduler_job_ids = {ti.queued_by_job_id for ti in tis if ti.queued_by_job_id}
+
+        # Tasks triggered through API will have no ti.queued_by_job_id
+        # and their pod will have label 'airflow-worker=manual'
+        if any(ti for ti in tis if not ti.queued_by_job_id):
```

Comment:
This change will fix the issue of tasks being incorrectly reset, but I think that in some cases we could still could end up with empty fields in the database, which might cause issues elsewhere or later on. 

Would it be possible instead to just set the `queued_by_job_id`  field to `manual` when the taskInstance is initially queued? We already have to manually write the `queued_dttm` for similar reasons. 

https://github.com/apache/airflow/blob/cfa95af7e83b067787d8d6596caa3bc97f4b25bd/airflow/www/views.py#L1811-L1819

Reply:
I am not sure which issues you are talking about, which makes it difficult for me to implement it. Also if you are speaking in general about queued_by_job_id not being filled in than that is an issue for any executor, so should that logic not reside in the scheduler code itself instead of in the kubernetes_executor?